devmgmtV2/controllers/upgrade.controller.js

```diff
@@ -30,8 +30,29 @@ let writeUpdateFile = (req, res) => {
   })
 }
 
+let checkPreviousVersion = (req,res) => {
+    let previousVersionPath = '/opt/opencdn.old'
+    if(fs.existsSync(previousVersionPath)) {
+        res.status(200).json({success : true});
+      } else {
+        res.status(200).json({success : false});
+      }
+}
 
+let revertVersion = (req, res) => {
+  exec('/opt/opencdn/CDN/revert.sh', (error, stdout, stderr) => {
+    if (error) {
+      res.status(200).json({success : false});
```

Comment:
You may want to add the console logs here? Logs are much important when things go wrong (i.e success=false)